LCM/scripts/PerformInventory.py

```diff
@@ -99,8 +101,8 @@ def usage():
     sys.exit(1)
 
 # Acquire inventory file lock
-filehandle = os.open(dsc_sysconfdir + "/inventory_lock", os.O_WRONLY | os.O_CREAT , 0o644)
-inventory_lock = os.fdopen(filehandle, 'w')
+inventory_lock = open(inventorylock_path, "w+")
+os.chmod(inventorylock_path , stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH)
```

Comment:
Mike Thompson brought up an issue on [my OMSAgent PR](https://github.com/Microsoft/OMS-Agent-for-Linux/pull/492) with this kind of approach: the file's permissions should be locked down as soon as it is created. With this change, you create the file and then change the permissions.

I would suggest checking if the file exists, and locking down permissions optimally depending on that: if the file exists, use chmod. Otherwise, create the file with the correct permissions.